superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterLabel.tsx

```diff
@@ -401,7 +401,7 @@ export default function DateFilterLabel(props: DateFilterControlProps) {
           ref={labelRef}
         />
       </Tooltip>
-      {/* the zIndex value is from trying so that the Modal doesn't overlay the AdhocFilter when GENERIC_CHART_AXES is enabled */}
+      {/* the zIndex value is from trying so that the Modal doesn't overlay the AdhocFilter */}
```

Comment:
Unrelated: Not sure I get the meaning of the sentence in this comment

Reply:
@geido The z-index value comes from _trying_, I'm not sure what's value is an appreciate value when I made this function.

Reply:
I think the developer meant that the z-index value was set, after multiple tentative values, to make sure the modal doesn't overlay the filter.

Agree it's confusing 🤷🏼‍♂️

Reply:
The problem here is that we don't have a z-index registry which would automatically resolve this type of issue. Maybe we'll have one some day.